cmd/infra/powervs/destroy.go

```diff
@@ -202,31 +230,37 @@ func (options *DestroyInfraOptions) DestroyInfra(ctx context.Context, infra *Inf
 	if !skipPowerVs {
 		session, err = createPowerVSSession(accountID, options.Region, options.Zone, options.Debug)
 		if err != nil {
-			return err
+			errL = append(errL, fmt.Errorf("error creating powervs session: %w", err))
+			log(options.InfraID).Error(err, "error creating powervs session")
 		}
 
-		if err = destroyPowerVsCloudConnection(ctx, options, infra, powerVsCloudInstanceID, session); err != nil {
-			errL = append(errL, fmt.Errorf("error destroying powervs cloud connection: %w", err))
-			log(options.InfraID).Error(err, "error destroying powervs cloud connection")
+		if session != nil {
+			if err = destroyPowerVsCloudConnection(ctx, options, infra, powerVsCloudInstanceID, session); err != nil {
+				errL = append(errL, fmt.Errorf("error destroying powervs cloud connection: %w", err))
+				log(options.InfraID).Error(err, "error destroying powervs cloud connection")
+			}
 		}
 	}
 
 	v1, err := createVpcService(options.VPCRegion, options.InfraID)
 	if err != nil {
-		return err
```

Comment:
lets make function for handling each service cleanup as mention in the previous comment.

Reply:
Addressed